opal/mca/btl/ofi/btl_ofi_frag.c

```diff
@@ -100,9 +101,29 @@ int mca_btl_ofi_send(mca_btl_base_module_t *btl, mca_btl_base_endpoint_t *endpoi
     /* This tag is the active message tag for the remote side */
     frag->hdr.tag = tag;
 
-    /* create completion context */
     context = get_ofi_context(ofi_btl);
-    comp = mca_btl_ofi_frag_completion_alloc(btl, context, frag, MCA_BTL_OFI_TYPE_SEND);
+    msg_sz = sizeof(mca_btl_ofi_header_t) + frag->hdr.len;
+
+    /*
+     * feeling lucky today, try inject path 
+     */
+    if (msg_sz <= mca_btl_ofi_component.max_inject_size &&
+        false == mca_btl_ofi_component.disable_inject) {
+        rc = fi_inject(context->tx_ctx, &frag->hdr, msg_sz, ofi_ep->peer_addr);
+        if (FI_SUCCESS == rc) {
+            mca_btl_ofi_frag_complete(frag, OPAL_SUCCESS);
+            return OPAL_SUCCESS;
+        }
+        /*
+         * -FI_EAGAIN is okay but any other error is a problem
```

Comment:
So the intent is that if `-FI_EAGAIN` happened, fallback to use `fi_send`?

If so, a comment would be nice.

Reply:
There's a note in the commit, but it'd be useful to have a comment in the code itself

Reply:
done